src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/GeneratedCodeContainer.cs

```diff
@@ -2,32 +2,88 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
-using Microsoft.CodeAnalysis.Text;
-using Microsoft.CodeAnalysis.Experiment;
-using Microsoft.AspNetCore.Razor.Language;
+using System.Collections.Generic;
 using System.Collections.Immutable;
-using System.Threading.Tasks;
 using System.Threading;
-using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.CodeAnalysis.Experiment;
+using Microsoft.CodeAnalysis.Text;
 
 namespace Microsoft.CodeAnalysis.Razor.ProjectSystem
 {
     internal class GeneratedCodeContainer : IDocumentServiceFactory, ISpanMapper
     {
+        public event EventHandler<TextChangeEventArgs> GeneratedCodeChanged;
```

Comment:
Allows people to grab the `sender` and look at some more information about the updated state.

Reply:
+Beans